libs/prelude/Language/Reflection/Elab.idr

```diff
@@ -83,6 +83,34 @@ record FunDefn a where
   clauses : List (FunClause a)
 
 
+||| A constructor to be associated with a new datatype.
+record ConstructorDefn where
+  constructor Constructor
+
+  ||| The name of the constructor. The name must _not_ be qualified -
+  ||| that is, it should begin with the `UN` or `MN` constructors.
```

Comment:
Why this limitation? You can define additionally-qualified data constructors in a normal data declaration, so it makes little sense to disallow it in Elab.